fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -4509,30 +4511,77 @@ private LocalDate getMaxDateLimitForNewRepayment(final PeriodFrequencyType perio
                                                    // startDate
     }
 
-    public void applyHolidayToRepaymentScheduleDates(final Holiday holiday) {
+    public void applyHolidayToRepaymentScheduleDates(final Holiday holiday, final LoanUtilService loanUtilService) {
+        final DefaultScheduledDateGenerator scheduledDateGenerator = new DefaultScheduledDateGenerator();
+        ScheduleGeneratorDTO scheduleGeneratorDTO = loanUtilService.buildScheduleGeneratorDTO(this, holiday.getFromDateLocalDate());
+        final LoanApplicationTerms loanApplicationTerms = this.constructLoanApplicationTerms(scheduleGeneratorDTO);
+
+        // LocalDate rescheduleFromDate = holiday.getFromDateLocalDate();
+        LocalDate adjustedRescheduleToDate = null;
+
+        if (holiday.getReScheduleType().isResheduleToNextRepaymentDate()) {
+            final LocalDate rescheduleToDate = holiday.getToDateLocalDate();
+            for (final LoanRepaymentScheduleInstallment loanRepaymentScheduleInstallment : this.getRepaymentScheduleInstallments()) {
+                if (rescheduleToDate.isEqual(loanRepaymentScheduleInstallment.getDueDate())) {
+                    adjustedRescheduleToDate = rescheduleToDate;
+                    break;
+                } else {
+                    // Account for a Bimonthly Loan Product
+                    // if (loanApplicationTerms.getRepaymentPeriodFrequencyType().isSemiMonthly()) {
+                    // if (rescheduleToDate.isAfter(loanRepaymentScheduleInstallment.getDueDate())
+                    // && rescheduleToDate.isBefore(loanRepaymentScheduleInstallment.getDueDate().plusDays(15))) {
```

Comment:
Why are you commenting all these lines if its not needed maybe we remove them? is it important?

Reply:
The commented code is more for a future proof for an upcoming feature that I think is still WIP progress at Muellners around a bi-monthly product.
The commented code is meant to retrofit the holiday rescheduling to new feature once it's rolled out.

Reply:
thanks Kaze for focusing on detail here. very thoughtful!